shopify-client/src/views/ShopList.vue

```diff
@@ -0,0 +1,261 @@
+<template>
+  <div class="shopList">
+    <h1>Shop List</h1> <!--Temp header-->
+    <div id="filters">
+      <div class="filterTile">
+        <h4>Categories</h4>
+        <div v-if="categories.length < 1">
+          <p>No available categories</p>
+        </div>
+        <div
+            v-else
+            v-for="category in categories"
+            :key="category.id"
+        >
+          <label :for="category.name">{{category.name}}</label>
+          <input
+              type="checkbox"
+              :id="category.name"
+              class="checkBox"
+          >
+        </div>
+      </div>
+      <div class="filterTile">
+        <h4>Tags</h4>
+        <div v-if="tags.length < 1">
+          <p>No available tags</p>
+        </div>
+        <div
+            v-else
+            v-for="tag in tags"
+            :key="tag.id"
+        >
+          <label :for="tag.name">{{tag.name}}</label>
+          <input
+              type="checkbox"
+              :id="tag.name"
+              class="checkBox"
+          >
+        </div>
+      </div>
+      <input
+          id="filterButton"
+          type="button"
+          value="Filter Search"
+          v-on:click="filterSearch"
+      />
+    </div>
+    <div id="filteredList">
+      <h4 id="searchKeyword">Search keyword: "{{keyword}}"</h4>
+      <div v-if="shops.length < 1">
+        No results found
+      </div>
+      <div
+          v-else
+          v-for="shop in shops"
+          :key="shop.id"
+          class="shopTile"
+      >
+        <img :src="'/' + shop.image"/> <!--Update when root image location is known -->
+        <h3>
+          <a :href="'/shop/' + shop.id">
+            {{shop.name.toLocaleUpperCase()}}
+          </a>
+        </h3>
+        <p>
+          {{shop.description}}
+        </p>
+        <div
+            v-for="category in shop.categories"
+            :key="category.id"
+            :class="'shopAttribute shopCategory ' + category.name"
+        >
+          {{category.name}}
+        </div>
+        <div
+            v-for="tag in shop.tags"
+            :key="tag.id"
+            :class="'shopAttribute shopTag ' + tag.name"
+        >
+          {{tag.name}}
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+//Placeholders
+//import header from '@/components/header.vue'
+//import footer from '@/components/footer.vue'
+
+let addDiff = (newAttributes, target) =>
+    newAttributes.forEach(function (attribute){
+      if(target.every(element => attribute.name !== element.name)){
+        target.push(attribute)
+      }
+    });
+
+//checker if the shop has the selected tag and categories
+let checker = (shop, target) =>
+    target.every(attribute => shop.getElementsByClassName(attribute).length > 0);
+
+export default {
+  name: "stopList",
+  data(){
+    return{
+      shops:[],
+      categories: [],
+      tags: [],
+      keyword: '',
+    }
```

Comment:
Define an `error` property here and display it when it is not empty.  Use the alerts classes https://getbootstrap.com/docs/4.0/components/alerts/